tablo/interfaces/arcgis/views.py

```diff
@@ -259,44 +266,61 @@ def handle_request(self, request, **kwargs):
         except (DatabaseError, ValueError):
             return HttpResponseBadRequest(json.dumps({'error': 'Invalid request'}))
 
+        # Process limited query before calculating totals and counts
+        exceeded_limit = len(query_response) > int(limit)
+        query_response = query_response[:-1] if exceeded_limit else query_response
+
         if return_count_only:
-            response = query_response[0]
+            data = query_response[0]
+        elif return_format == 'csv':
+            header = list(query_response[0].keys())
+            header.sort()
```

Comment:
This has been taken care of in the latest push: the ordering will be the same as specified in the fields in the request parameter, and if `*` is expanded to a full set of fields, the order of the expanded fields matches the ordering of the columns in the database.